src/tests/test_bresenham.py

```diff
@@ -0,0 +1,75 @@
+import pytest
+
+from src.bresenham import bresenham
+
+
+@pytest.mark.parametrize(
+    ("x1", "y1", "x2", "y2", "result"),
+    (
+        (0, 0, 0, 0, ((0, 0),)),
+        (0, 0, 5, 0, ((0, 0), (1, 0), (2, 0), (3, 0), (4, 0), (5, 0))),
+        (0, 0, -5, 0, ((0, 0), (-1, 0), (-2, 0), (-3, 0), (-4, 0), (-5, 0))),
+        (0, 0, 0, 5, ((0, 0), (0, 1), (0, 2), (0, 3), (0, 4), (0, 5))),
+        (0, 0, 0, -5, ((0, 0), (0, -1), (0, -2), (0, -3), (0, -4), (0, -5))),
+        (0, 0, 2, 3, ((0, 0), (1, 1), (1, 2), (2, 3))),
+        (0, 0, -2, 3, ((0, 0), (-1, 1), (-1, 2), (-2, 3))),
+        (0, 0, 2, -3, ((0, 0), (1, -1), (1, -2), (2, -3))),
+        (0, 0, -2, -3, ((0, 0), (-1, -1), (-1, -2), (-2, -3))),
+        (-1, -3, 3, 3, ((-1, -3), (0, -2), (0, -1), (1, 0), (2, 1), (2, 2), (3, 3))),
+        (
+            0,
+            0,
+            11,
+            1,
+            (
+                (0, 0),
+                (1, 0),
+                (2, 0),
+                (3, 0),
+                (4, 0),
+                (5, 0),
+                (6, 1),
+                (7, 1),
+                (8, 1),
+                (9, 1),
+                (10, 1),
+                (11, 1),
+            ),
+        ),
+    ),
+)
```

Comment:
parametrising is very convenient and a good practice, but for the record, this is super hard to read

Reply:
Original code is not mine, as I said I pulled in from the same place @louis-e pulled the functionality.
The way it is formatted is just black follow its rules. 

I would probably prefer to do the whole set of parameters as a list of dicts, didn't seem worth it to change to my normal style since I was just copying the code anyway. But I might if you found it hard enough to read to be worth commenting on.